src/components/SearchBar.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { view , store } from 'react-easy-state';
+import { BrowserRouter as Router, Redirect, Link } from "react-router-dom"
+import './Searchbar.css';
+
+interface SearchProps {
+    type:string,
+    placeholder?:string
+}
+
+interface SearchState {
+    searchValue?: string,
+    redirect?: boolean
+}
+
+const state = store<SearchState>({})
+
+class SearchBar extends React.Component<SearchProps>{
+    checkSubmit = (e:any) => {
+        if(e && e.keyCode == 13) {
+            state.redirect = true;
+        }
+    }
+
+    onChange = (e:any) => {
```

Comment:
You don't need to use the `=` here.

It's much nicer to use:
```typescript
class ... {
    onChange(ev:any) {
    
    }
}
```

Also, the convention here is to name event variables `ev`, rather than `e`